### PR TITLE
Fully support use of BATTERY items by tools

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Infrastructure "Fully support use of BATTERY items by tools"

#### Purpose of change
#32067

#### Describe the solution
- [x] Update `item::info` for new `islot_tool` data
- [x] Specify compatible batteries in tool's definition 
- [x] Validate tool definitions in `Item_factory::check_definitions`
- [x] Allow reloading tools with `BATTERY` items
- [ ] Allow consumption of charge in a `BATTERY` item
- [ ] Document new JSON

#### Additional context
Depends on the new `BATTERY` item implementation in #32141. This feature branch is based on `battery` and should only be merged after #32141.

### CODE REVIEW

File: src/item_factory.cpp [hunk 1]
```diff
@@ -1888,6 +1888,21 @@ void Item_factory::load_magazine( JsonObject &jo, const std::string &src )
     }
 }
 
+void Item_factory::load( islot_battery &slot, JsonObject &jo, const std::string & )
```
Comment (NickGrease) [inline src/item_factory.cpp]: Should it be `const std::string &src` ?

Comment (ifreund) [reply src/item_factory.cpp]: That would actually cause a compiler warning since the `const std::string &` parameter isn't used.

Comment (NickGrease) [reply src/item_factory.cpp]: I'll just stick upvoting - I know I was stepping out of my bounds :P

Comment (ifreund) [reply src/item_factory.cpp]: No worries!

**Pull request closed without merging**